src/features/user/DonationReceipt/microComponents/DonorContactForm.tsx

```diff
@@ -0,0 +1,280 @@
+            tenant: tenantConfig.id,
+            url: '/app/profile',
+            data: body,
+            token,
+            logoutUser,
+          });
+
+          if (!updatedUser) return;
+          setUser(updatedUser);
+        }
+        const {
+          amount,
+          currency,
+          donations,
+          dtn,
+          year,
+          challenge,
+          downloadUrl,
+          verificationDate,
+        } = donationReceiptData;
+        // The donor can only modify their name, TIN, and address in the Donor Contact Management page.
+        // The rest of the donation receipt data remains unchanged.
+        const donorDetails = getUpdatedDonorDetails(
+          updatedUser,
+          checkedAddressGuid
+        );
+        updateDonationReceiptData({
+          donor: {
+            ...donorDetails,
+            type:
+              updatedUser.type === 'individual' ? 'individual' : 'organization',
+            guid: checkedAddressGuid,
+          },
+          hasDonorDataChanged: true,
+          amount,
+          currency,
+          donations,
+          dtn,
+          year,
+          challenge,
+          downloadUrl,
+          verificationDate,
+        });
+        navigateToVerificationPage();
+      } catch (err) {
+        setErrors(handleError(err as APIError));
+        setIsLoading(false);
+      }
+    },
+    [
+      user,
+      token,
+      contextLoaded,
+      isUserDataChanged,
+      tenantConfig.id,
+      logoutUser,
+      setUser,
+      checkedAddressGuid,
+      navigateToVerificationPage,
+      updateDonationReceiptData,
+      setErrors,
+    ]
+  );
+
+  return (
+    <form className={styles.donorContactForm}>
+      <InlineFormDisplayGroup>
+        {user.type === 'organization' && (
+          <FormInput
+            name={'companyName'}
+            control={control}
+            rules={{ required: t('donorInfo.companyRequired') }}
+            label={t('donorInfo.companyName')}
+          />
+        )}
+        {user.type === 'individual' && (
+          <InlineFormDisplayGroup>
+            <FormInput
+              name={'firstName'}
+              control={control}
+              rules={{ required: t('donorInfo.firstNameRequired') }}
+              label={t('donorInfo.firstName')}
+            />
+            <FormInput
+              name={'lastName'}
+              control={control}
+              rules={{ required: t('donorInfo.lastNameRequired') }}
+              label={t('donorInfo.lastName')}
+            />
+          </InlineFormDisplayGroup>
+        )}
+        {user.tin !== null && (
+          <FormInput
+            name={'tin'}
+            control={control}
+            rules={{ required: t('donorInfo.tinRequired') }}
+            label={t('donorInfo.tin')}
+          />
+        )}
+      </InlineFormDisplayGroup>
+
+      <section className={styles.donorAddressSection}>
+        {donorAddresses.map((address) => {
```

Comment:
Consider whether `donorAddresses` is needed, as the `user` will have updated address information.